src/PerfView/GcStats.cs

```diff
@@ -830,6 +830,45 @@ private static void PrintEventCondemnedReasonsTable(TextWriter writer, TraceProc
                 return;
             }
 
+            List<TraceGC> events = new List<TraceGC>();
+            List<byte[]> condemnedReasonRows = new List<byte[]>();
+            for (int i = start; i < runtime.GC.GCs.Count; i++)
+            {
+                var _event = runtime.GC.GCs[i];
+                if (filter == null || filter(_event))
+                {
+                    if (!_event.IsComplete)
+                    {
+                        continue;
+                    }
+                }
+                events.Add(_event);
+                condemnedReasonRows.Add(GetCondemnedReasonRow(_event));
+            }
+
+            bool hasAnyContent = false;
+            bool[] columnHasContent = new bool[CondemnedReasonsHtmlHeader.Length];
+            foreach (byte[] condemnedReasonRow in condemnedReasonRows)
+            {
+                for (int j = 0; j < CondemnedReasonsHtmlHeader.Length; j++)
+                {
+                    if (columnHasContent[j])
+                    {
+                        break;
+                    }
+                    if (condemnedReasonRow[j] != 0)
+                    {
+                        hasAnyContent = true;
+                        columnHasContent[j] = true;
+                    }
+                }
+            }
+
+            if (!hasAnyContent)
```

Comment:
we'd still want to display the text that says ""Condemned reasons for GC" from below when there's no condemned reasons because this is saying "we do have the events to decode the condemned reasons but there's none" vs. "we don't have the events to decode the condemned reasons and we will not even display the text".